### PR TITLE
migrate v1alpha4 to v1beta1

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
1. Add v1beta1 cluster apis
2. Add conversion-gen tool support
3. Rename APIEndpoint to VPCEndpoint --> conversion-gen autogenerated code will generate a conflicting conversion function for cluster-api(APIEndpoint) reference and cluster-api-provider-ibmcloud(APIEndpoint). Hence to avoid conflicting functions which prevents autogeneration of conversion code renaming cluster-api-provider-ibmcloud(APIEndpoint) to VPCEndpoint
3. Add v1alpha4 to v1beta1 conversion

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by mkumatag** [comment]
We should convert the other IBM Cloud APIs as well,

/ok-to-test

**Comment by mkumatag** [comment]
need some more changes before this more info - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/446

File: hack/tools/Makefile [hunk 1]
```diff
@@ -74,3 +74,7 @@ $(MDBOOK_TABULATE): $(BIN_DIR) go.mod go.sum
 GOJQ := $(BIN_DIR)/gojq
 $(GOJQ): $(BIN_DIR) go.mod go.sum
 	go build -tags=tools -o $@ github.com/itchyny/gojq/cmd/gojq
+
+CONVERSION_GEN := $(BIN_DIR)/conversion-gen
+$(CONVERSION_GEN): $(BIN_DIR) go.mod go.sum
+	go build -tags=tools -o $(BIN_DIR)/conversion-gen k8s.io/code-generator/cmd/conversion-gen
```
Comment (mkumatag) [inline hack/tools/Makefile:80]: can follow same pattern like others in this file
```suggestion
	go build -tags=tools -o $@ k8s.io/code-generator/cmd/conversion-gen
```

File: config/samples/infrastructure_v1beta1_ibmpowervscluster.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: IBMPowerVSCluster
+metadata:
+  name: ibmpowervscluster-sample
+spec:
+  # TODO(user): Add fields here
```
Comment (mkumatag) [inline config/samples/infrastructure_v1beta1_ibmpowervscluster.yaml:6]: wondering if we can add something here, this comment applies for the remaining samples as well.

File: api/v1beta1/ibmpowervscluster_types.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+// IBMPowerVSClusterStatus defines the observed state of IBMPowerVSCluster
+type IBMPowerVSClusterStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	Ready bool `json:"ready"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
```
Comment (mkumatag) [inline api/v1beta1/ibmpowervscluster_types.go:58]: I see `v1beta1` is set as storageversion, so wondering if we need to remove storageversion for other version.

**Review by mkumatag — CHANGES_REQUESTED** [review]
Lets run one round of tests and see how it goes as well :)

File: api/v1beta1/ibmpowervscluster_types.go (hunk 3 above)
Comment (Prajyot-Parab) [reply api/v1beta1/ibmpowervscluster_types.go:58]: This is already taken care of.

File: hack/tools/Makefile (hunk 1 above)
Comment (Prajyot-Parab) [reply hack/tools/Makefile:80]: Done working fine.

**Review by mkumatag — CHANGES_REQUESTED** [review]
we should also bump the cluster-api module to the latest level here - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/937094ee72524591e6c232e7537ce83de355f221/go.mod#L25

File: config/samples/infrastructure_v1beta1_ibmpowervscluster.yaml (hunk 2 above)
Comment (Prajyot-Parab) [reply config/samples/infrastructure_v1beta1_ibmpowervscluster.yaml:6]: Done

**Comment by mkumatag** [comment]
Also, need an entry here:

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/937094ee72524591e6c232e7537ce83de355f221/metadata.yaml#L10

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/937094ee72524591e6c232e7537ce83de355f221/config/crd/kustomization.yaml#L5

File: api/v1alpha3/ibmvpccluster_types.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ package v1alpha3
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
Comment (mkumatag) [inline api/v1alpha3/ibmvpccluster_types.go:21]: should be
```golang
clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
```

File: api/v1alpha4/ibmpowervscluster_types.go [hunk 5]
```diff
@@ -18,7 +18,7 @@ package v1alpha4
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
Comment (mkumatag) [inline api/v1alpha4/ibmpowervscluster_types.go:21]: should be
```golang
clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
```

File: api/v1alpha4/ibmvpccluster_types.go [hunk 6]
```diff
@@ -18,7 +18,7 @@ package v1alpha4
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
Comment (mkumatag) [inline api/v1alpha4/ibmvpccluster_types.go:21]: should be:
```golang
clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
```

File: metadata.yaml [hunk 7]
```diff
@@ -5,6 +5,9 @@
 # update this file only when a new major or minor version is released
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 releaseSeries:
+  - major: 0
+    minor: 1
```
Comment (mkumatag) [inline metadata.yaml:9]: this will be 2

**Review by mkumatag — CHANGES_REQUESTED** [review]
few minor comments

File: api/v1alpha3/ibmvpccluster_types.go [hunk 8]
```diff
@@ -60,7 +60,7 @@ type IBMVPCClusterStatus struct {
 	// Bastion Instance `json:"bastion,omitempty"`
 	Ready       bool        `json:"ready"`
 	Subnet      Subnet      `json:"subnet,omitempty"`
-	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
+	VPCEndpoint VPCEndpoint `json:"vpcEndpoint,omitempty"`
```
[the remark this answers is not in the document]
Comment (mkumatag) [reply api/v1alpha3/ibmvpccluster_types.go:63]: Seem not, I don't see atleast any references in the hypershift project, hence merging this PR

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**